Sources/Brave/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2411,8 +2411,11 @@ public class BrowserViewController: UIViewController {
 
 extension BrowserViewController {
   func didScanQRCodeWithURL(_ url: URL) {
-    popToBVC()
-    finishEditingAndSubmit(url, visitType: .typed)
+    let overlayText = URLFormatter.formatURL(url.absoluteString, formatTypes: [], unescapeOptions: [])
+
+    popToBVC() {
+      self.topToolbar.enterOverlayMode(overlayText, pasted: false, search: false)
+    }
 
     if !url.isBookmarklet && !PrivateBrowsingManager.shared.isPrivateBrowsing {
       RecentSearch.addItem(type: .qrCode, text: nil, websiteUrl: url.absoluteString)
```

Comment:
Oh just realized this, before we merge: does this logic change? Do we consider a URL that isn't autosubmitted now part of a recent search list? cc @iccub

Reply:
I was actually considering removing `RecentSearch.addItem`, but decided it's still useful to add scanned URLs to search history. Just in case user would want to visit it later.